flow/connectors/s3/s3.go

```diff
@@ -0,0 +1,118 @@
+package conns3
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/PeerDB-io/peer-flow/connectors/utils"
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/model"
+	"github.com/aws/aws-sdk-go/service/s3"
+	log "github.com/sirupsen/logrus"
+)
+
+type S3Connector struct {
+	url    string
+	client s3.S3
+}
+
+func NewS3Connector(ctx context.Context,
+	S3ProtoConfig *protos.S3Config) (*S3Connector, error) {
+	s3Client, err := utils.CreateS3Client()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create S3 client: %w", err)
+	}
+	return &S3Connector{
+		url:    S3ProtoConfig.Url,
+		client: *s3Client,
+	}, nil
+}
+
+func (c *S3Connector) Close() error {
+	log.Infof("Close not needed for S3")
+	return nil
+}
+
+func (c *S3Connector) ConnectionActive() bool {
+	_, err := c.client.ListBuckets(nil)
+	return err == nil
+}
+
+func (c *S3Connector) NeedsSetupMetadataTables() bool {
```

Comment:
for all the non-qrep methods (CDC methods like most of these in this file, can you return error:

`cdc based replication is not currently supported for S3 target`.